common/status.py

```diff
@@ -0,0 +1,331 @@
+                )
+
+        def remove_keys(dic, keys):
+            """Helper function to remove specified keys from a dict."""
+            if isinstance(dic, dict):
+                return {
+                    key: remove_keys(value, keys) for key, value in dic.items()
+                    if not self.all_status or key not in keys
+                }
+
+            return dic
+
+        # Fields to remove if returning list of statuses
+        keys_to_remove = ['Backup mode', 'Paths']
+
+        result = {
+            key: remove_keys(value, keys_to_remove)
+            for key, value in self.status.items()
+            if (self.all_status or profile_filter(key)) and issues_filter(key)
+        }
+
+        if self.json:
+            return json.dumps(result, indent=2)
+
+        return self._human(result) + '\n\n'
+
+    def _date_of(self, filename):
+        file_path = pathlib.Path(filename)
+        try:
+            timestamp = file_path.stat().st_mtime
+            timestamp_ts = datetime.fromtimestamp(timestamp)
+            return timestamp_ts.strftime("%Y-%m-%d %H:%M:%S")
+
+        except FileNotFoundError:
+            return None
+
+    def _longest_key(self, dic, depth=0):
+        """
+        Get the length of the longest key from a dict (dic) to assist
+        with formatting the output.
+        """
+        if depth > 0:
+            max_len = max(map(len, dic.keys()), default=0)
+        else:
+            max_len = 0
+
+        for value in dic.values():
+            if isinstance(value, dict):
+                max_len = max(
+                            max_len,
+                            self._longest_key(value, depth=depth + 1)
+                            )
+
+        return max_len
+
+    def _human(self, dic, indent=0, width=None):
+        """
+        Return dict (dic) as a human readable formatted string.
+        """
+        human = []
+        singly_nested = 2
+        doubly_nested = 4
+
+        if width is None:
+            width = self._longest_key(dic) + 1
+        if indent == 0:
+            human.append('')
+
+        for key, value in dic.items():
+            if indent != doubly_nested:
+                human.append('')
+
+            if isinstance(value, dict):
+                human.append(f"{' ' * indent}{key}:")
+                human.append(
+                    self._human(value, indent=indent + 2, width=width)
+                    )
+
+            elif indent == singly_nested:
+                human.append(f"{' ' * indent}{key:{width + 2}}: {value}")
+
+            else:
+                human.append(f"{' ' * indent}{key:{width}}: {value}")
+
+        return '\n'.join(human)
+
+    def _file_path(self):
+        xdg_state = os.environ.get('XDG_STATE_HOME', None)
+        if xdg_state:
+            xdg_state = Path(xdg_state)
+        else:
+            xdg_state = Path.home() / '.local' / 'state'
+
+        return xdg_state / 'backupstatus.json'
```

Comment:
```suggestion
    def _file_path(self):
        xdg_state = os.environ.get(
            'XDG_STATE_HOME',
            Path.home() / '.local' / 'state')

        return Path(xdg_state) / bitbase.BINARY_NAME_CLI / 'backupstatus.json'
```

`bitbase.BINARY_NAME_CLI` is "backintime". Don't forget to `import bitbase` on top of the file.

I know it looks akward, because in case `XDG_STATE_HOME` is not present, the default value is of type `Path` and because of that, in the `return` statement, `Path()` will be initialized with that default `Path` instance. But it works.

Reply:
is this a new location? On my machine, the state files are in ~/.local/state

Reply:
Ah, I see. You are right, currently there is not dedicated sub-folder for BIT related state files. And I think we don't need it.

In this case I would recommend to modify the name of your state file but not the path.

    Path.home(xdg_state) / 'backintime-status.json'

You can mke the name longer and more informative if you want, e.g. `backintime-backup-status.json`. That files are not noticed or read by human beings in most cases.